source/documentation/airflow.md

```diff
@@ -333,10 +346,75 @@ To deploy your Airflow sandbox, follow the instructions in the [Work with Analyt
+
+- **airflow_my_task is not authorized to perform: iam:PassRole on resource: arn:aws:iam::593291632749:role/airflow_my_task** The task IAM policy needs to have gluejobs enabled.
+
+- **s3 credentials error** IAM policy needs to allow appropriate read/write access to the s3 resource you are accessing.
+
+- **AWS Athena timeout error** If a task *intermittently* fails when reading data with Athena, it may help to pass the env_args `"AWS_METADATA_SERVICE_TIMEOUT": "60"` and `"AWS_METADATA_SERVICE_NUM_ATTEMPTS": "5"` to make your pod wait longer for Athena (in seconds) and make retries in the event of failure.
+
+- **Other errors once pod is running** Likely to be Python, R or Spark errors from within the task source code.
+
+## Notes on creating Docker environments
+
+The following are the essential aspects in a Dockerfile
+
+### Configuration of software to be installed
+
+There are a couple of parts to this. The first is the basic foundation image to work with. This encompasses the operating system and basic software, like python. This is defined in the `FROM` part of a Dockerfile, eg. `FROM python:3.7`. 
+<details>
+<summary>You should select this from a list available on the AP.</summary>
+
+```
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:3.7
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:3.7-slim
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:alpine3.10
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:slim
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:3.7-slim-buster
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:alpine3.9
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:3.6
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/python:3.6-slim
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/r-base:3.5.0
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/rocker:shiny-3.4.2
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/amancevice:pandas-0.25.2
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/moj-spark-jovyan:baseenv
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/nginx-unprivileged:stable-alpine
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/continuumio:miniconda3
+593291632749.dkr.ecr.eu-west-1.amazonaws.com/pottava:s3-proxy
+```
+
+</details>
+Additionally you will want to install specific Python or R packages that you use in your code. You can do this with `RUN pip install -r requirements.txt` as referenced above. To be robust, you should pin all versions (`package==1.2.0`). For complicated installations this is best done by activating a conda environment:
```

Comment:
Is our position to still recommend using conda, given all the problems it causes?

Reply:
As I understand it using conda is unavoidable in some circumstances. It would benefit from noting what those circumstances are, but I'd need to make sure I know what I'm talking about first.